### PR TITLE
Use shared config and eslint

### PR DESCRIPTION
**Pull request by karelhala**
### Shared config

Since we have fairly robust shared config on plaform let's use it. This PR also adds shared eslint config so we don't forget about keys and such.

### Share routes trough fed modules

With the shared config we can now expose these routes over federated modules and don't have to use npm for that. The RBAC work is here https://github.com/RedHatInsights/insights-rbac-ui/pull/783

### CODE REVIEW

**Comment by karelhala** [comment]
I wanted to use webpack proxy for this app as wel so we don't have to use just insights-proxy, but I had hard time making it run @rvsia do you think you could look into it once this PR is merged?

File: config/prod.webpack.config.js
Comment (Hyperkid123) [inline config/prod.webpack.config.js:13]: You are not exposing the same files as in the dev config.

File: package.json
Comment (Hyperkid123) [inline package.json:14]: `stylelint` is not in dependencies (I don't think we need it IMO)

File: src/App.js
Comment (Hyperkid123) [inline src/App.js:14]: Can you remove the withRouter decorator and use the history hook?

File: config/prod.webpack.config.js
Comment (karelhala) [reply config/prod.webpack.config.js:13]: Good catch!

**Review by Hyperkid123 — APPROVED** [review]

File: src/App.js
Comment (redallen) [inline src/App.js:30]: If you prefer `useHistory` to `withRouter` it'd be good to change the starter app.

**Review by redallen — CHANGES_REQUESTED** [review]
I'm not against `eslint`, sharing pages via federated modules instead of NPM, or using `@redhat-cloud-services/frontend-components-config-utilities`. However, I am against removing `@redhat-cloud-services/insights-standalone` since it provides a better dev experience and this is the test app I use for it!

File: src/App.js
Comment (karelhala) [reply src/App.js:30]: Good point. We should go trough the starter app and clean the code there.

File: package.json
Comment (karelhala) [reply package.json:64]: Sure thing!

File: src/Components/AccessRequestsTable.js
Comment (karelhala) [reply src/Components/AccessRequestsTable.js:1]: Yeah, I tried to fix as much issues as possible, but let's make this in a proper way.

File: src/Routes/AccessRequestDetailsPage.js
Comment (karelhala) [reply src/Routes/AccessRequestDetailsPage.js:29]: It uses `propTypes` if we change the prop structure it will tell us right away.

File: src/Routes/AccessRequestsPage.js
Comment (karelhala) [reply src/Routes/AccessRequestsPage.js:8]: It uses `propTypes` if we change the prop structure it will tell us right away.

**Comment by karelhala** [comment]
@redallen addressed your feedback. Mind taking another look?

File: src/Routes/AccessRequestDetailsPage.js
Comment (redallen) [reply src/Routes/AccessRequestDetailsPage.js:29]: Sure, but `propTypes` doesn't tell you _why_. A comment does.

**Comment by redallen** [comment]
This is good except for removing `insights-standalone` which I'd like to demo in this app on Thursday. It can be merged after and then another PR (or this PR) can re-adopt `insights-standalone`.

**Review by redallen — APPROVED** [review]

**Pull request merged**
